src/db/commands.rs

```diff
@@ -417,3 +434,30 @@ fn handle_user_result(
     user.connected_at = connected_at;
     Ok(user)
 }
+
+/// Like Result::ok, convert from Result<T, E> to Option<T> but applying a
+/// function to the Err value
+fn ok_or_inspect<T, E, F>(result: StdResult<T, E>, op: F) -> Option<T>
+where
+    F: FnOnce(E),
+{
+    match result {
+        Ok(t) => Some(t),
+        Err(e) => {
+            op(e);
+            None
+        }
+    }
+}
+
+/// Log/metric errors during conversions to Notification
+fn conversion_err<E>(metrics: &StatsdClient, err: E, name: &'static str)
+where
+    E: Display,
+{
+    error!("Failed {} conversion: {}", name, err);
```

Comment:
I had this as debug! but figured let's error! for now so it's seen on prod

Reply:
:+1: